### PR TITLE
Common-crafting update.

### PR DESCRIPTION
**Pull request by Dartellum**

### CODE REVIEW

File: common-crafting.lic [hunk 1]
```diff
@@ -181,4 +181,38 @@ module DRCC
       DRC.bput("stow my #{logbook} logbook", 'You put', 'What were you referring to')
     end
   end
+
+  def order_enchant(stock_room, stock_needed, stock_number)
+    stock_needed.times do
+      order_item(stock_room, stock_number)
+      stow_crafting_item(left_hand, @bag, @belt)
```
Comment (rpherbig) [inline common-crafting.lic]: ❓ Will `@bag` and `@belt` resolve correctly? I think you'll need to make them function args

File: common-crafting.lic [hunk 2]
```diff
@@ -181,4 +181,38 @@ module DRCC
...
+      stow_crafting_item(right_hand, @bag, @belt)
+      next unless left_hand && right_hand
+    end
+  end
+
+  def fount(stock_room, stock_needed, stock_number, quantity)
```
Comment (rpherbig) [inline common-crafting.lic]: 💡 Could add a verb to this function name? What is it doing with a fount?

File: common-crafting.lic [hunk 3]
```diff
@@ -181,4 +181,38 @@ module DRCC
...
+    case bput('tap my fount', /You tap .* inside your .*/, /You tap .*your .*/, /I could not find what you were referring to./)
```
Comment (rpherbig) [inline common-crafting.lic]: ❓ Would using `DRCI.exists?` or `DRCI.inside?` simplify this?

File: common-crafting.lic (hunk 1 above)
Comment (Dartellum) [reply common-crafting.lic]: The resolve for me. stow_crafting_item is a function in common-crafting.

File: common-crafting.lic (hunk 2 above)
Comment (Dartellum) [reply common-crafting.lic]: This function checks for the fount, either on a belt, in a container, or on a brazier. Then, the fount count is found and will discard if below a certain threshold and purchase a new one.

File: common-crafting.lic (hunk 3 above)
Comment (Dartellum) [reply common-crafting.lic]: I would need to test.

File: common-crafting.lic (hunk 1 above)
Comment (rpherbig) [reply common-crafting.lic]: I think you've lucked into some state in which those happen to be populated. This is probably due to some nuance of how import works and we shouldn't rely on that. Using function args is safer.

File: common-crafting.lic [hunk 4]
```diff
@@ -181,4 +181,38 @@ module DRCC
...
+
+  def fount(stock_room, stock_needed, stock_number, quantity, bag, belt)
+    case bput('tap my fount', /You tap .* inside your .*/, /You tap .*your .*/, /I could not find what you were referring to./)
+    when /You tap (.*) inside your (.*)/, /You tap (.*) your (.*)/
+      /(\d+)/ =~ bput('analyze my fount', 'This appears to be a crafting tool and it has approximately \d+ uses remaining')
+      if Regexp.last_match(1).to_i < (quantity + 1)
+        get_tool('fount')
+        dispose('fount')
+        stow_hands
+        order_enchant(stock_room, stock_needed, stock_number)
```
Comment (rpherbig) [inline common-crafting.lic]: ❌ These three calls to `order_enchant` also need `bag` and `belt`

Comment (Dartellum) [reply common-crafting.lic]: Added.

**Pull request merged**